runtime/vm/jvminit.c

```diff
@@ -993,7 +1008,14 @@ freeJavaVM(J9JavaVM * vm)
 	}
 #endif /* JAVA_SPEC_VERSION >= 22 */
 
-	j9mem_free_memory(vm);
+	if (!IS_SNAPSHOTTING_ENABLED(vm)) {
+		j9mem_free_memory(vm);
+#if defined(J9VM_OPT_SNAPSHOTS)
+	} else {
+		VMSnapshotImplPortLibrary *imagePortLibrary = vm->vmSnapshotImplPortLibrary;
+		shutdownVMSnapshotImpl(imagePortLibrary);
+#endif /* defined(J9VM_OPT_SNAPSHOTS) */
+	}
```

Comment:
Again, an earlier version was better:
```c
#if defined(J9VM_OPT_SNAPSHOTS)
	if (IS_SNAPSHOTTING_ENABLED(vm)) {
		VMSnapshotImplPortLibrary *imagePortLibrary = vm->vmSnapshotImplPortLibrary;
		shutdownVMSnapshotImpl(imagePortLibrary);
	} else
#endif /* defined(J9VM_OPT_SNAPSHOTS) */
	{
		j9mem_free_memory(vm);
	}
```

Reply:
I agree. I'd thought one of your previous concerns was the hanging `{...}` blocks and splitting up the `} else {`, but I guess that was just in a few cases.

I'll go through and revert some of these to what I had before.